### ISSUE TITLE
RECALL and MRR very low 

### ISSUE THREAD

**Issue by eveTu** [pcerdam/KerasGRU4Rec#2]
I've tested the gru4rec implementation on 150K fashion clicks sequences. 
I got very low MRR and Recall. (**0.001698**  RECALL and **0.000509** MRR ) 
batch size = 10 (because my seq length are small  

any advice can help me. 
Thanks
<img width="289" alt="Screen Shot 2020-04-28 at 9 35 45" src="https://user-images.githubusercontent.com/14997823/80455152-ec5bd480-8933-11ea-88a9-d888a5d7405f.png">

**Comment by pcerdam** [comment]
Hi there,

My first recomendation would be to try out your dataset on other implementations of GRU4Rec or similar recommendation architectures. I'm guessing your dataset might just be harder than the RSC15 dataset to recommend for, which would explain the low metric values.

Let me know how that goes, and thanks for your interest!

**Comment by eveTu** [comment]
Thanks for your quick response! 
Now I understand that the batch size is a parameter for the training part and not represent the session length. So, I divided my training set into 4 batches (80,000 sessions total)  and got better MRR and Recall **( recall: 0.08, mrr: 0.04)**  , But there is still room for improvement

About the prediction part: in real-time my "batch size" is 1 that contains different session lengths. 
My question is where the model takes into account the session length? 
I created a new model that takes the weights from the saved model (train_batch_size,1,n_items)  to be (1,1,n_items).  but It seems that every "session" takes only 1 item? (maybe I'm wrong) 

I really appreciate the help

**Comment by pcerdam** [comment]
This implementation follows directly what the original [paper](https://arxiv.org/abs/1511.06939) proposes. In practice, this means that the model effectively trains in a session-parallel manner. To take into account different session lengths, the model resets the recurrent hidden state whenever the session ID changes (check out line 236 in the ```gru4rec.py``` file).  

So yes, the "batch size" dimension actually represents the number of sessions that the model trains (or, in this case, recommends for) in parallel. If you only want to predict one session at a time, then the tensor will have shape ```(1, 1, n_items)```.
